bob/src/cleaner.rs

```diff
@@ -22,27 +24,74 @@ impl Cleaner {
             hard_open_blobs,
             bloom_filter_memory_limit,
             index_memory_limit,
+            index_cleanup_requested: Mutex::new(false),
         }
     }
 
-    pub(crate) fn spawn_task(&self, backend: Arc<Backend>) {
+    pub(crate) fn spawn_task(&self, cleaner: Arc<Cleaner>, backend: Arc<Backend>) {
         tokio::spawn(Self::task(
-            backend,
+            cleaner.clone(),
+            backend.clone(),
             self.old_blobs_check_timeout,
             self.soft_open_blobs,
             self.hard_open_blobs,
             self.bloom_filter_memory_limit,
-            self.index_memory_limit,
         ));
+        tokio::spawn(Self::fast_cleaner_task(
+            cleaner,
+            backend,
+            Duration::from_secs(30),
+            self.index_memory_limit
+        ));
+    }
+
+    pub(crate) async fn request_index_cleanup(&self) {
+        let mut lock = self.index_cleanup_requested.lock().await;
+        *lock = true;
+    }
+
+    async fn fast_cleaner_task(
+        cleaner: Arc<Cleaner>,
+        backend: Arc<Backend>,
+        t: Duration,
+        index_memory_limit: Option<usize>,
+    ) {
+        let mut interval = interval(t);
+        loop {
+            interval.tick().await;
+            if let Some(limit) = index_memory_limit {
+                let mut should_clean_up = false;
+                {
+                    let mut lock = cleaner.index_cleanup_requested.lock().await;
+                    if *lock {
+                        *lock = false;
+                        should_clean_up = true;
+                    }
+                }
+                if should_clean_up {
+                    let mut memory = backend.index_memory().await;
```

Comment:
I think, it would be good to have 2 memory limits: soft and hard. Here the hard limit fits best, and soft limit is better for usual clean-up task. 
From the config perspective, the current `index_memory_limit` is basically a hard limit. Additional new parameter (`index_memory_limit_soft`) should be added. This new parameter should be `Optional` and by default calculated as `0.9 * index_memory_limit`